src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -302,47 +305,103 @@ protected override SyntaxNode FindStatementAndPartner(SyntaxNode declarationBody
                 partner = null;
             }
 
-            while (node != declarationBody && !StatementSyntaxComparer.HasLabel(node) && !LambdaUtilities.IsLambdaBodyStatementOrExpression(node))
+            while (true)
             {
-                node = node.Parent!;
-                if (partner != null)
+                bool isBody = node == declarationBody || LambdaUtilities.IsLambdaBodyStatementOrExpression(node);
+
+                if (isBody || StatementSyntaxComparer.HasLabel(node))
                 {
-                    partner = partner.Parent;
-                }
-            }
+                    switch (node.Kind())
+                    {
+                        case SyntaxKind.Block:
+                            statementPart = (int)GetStatementPart((BlockSyntax)node!, position);
```

Comment:
I'm surprised you need to put `node!` here. It has to be none null from the call to `node.kind()` in the switch :(

Reply:
It's actually not needed. It might have been annotated before we had `[NotNull]` on the node parameter.